projects/nlp/hello_classification_model.py

```diff
@@ -73,89 +84,129 @@ def instantiateModel(args):
   # Create model after setting specific arguments required for this experiment
   args.networkConfig = getNetworkConfig(args.networkConfigPath)
   args.k = kValues.get(args.modelName, 1)
-  args.numLabels = 2
-  model = createModel(**vars(args))
 
-  return model
+  return createModel(**vars(args))
 
 
 
-def trainModel(model, trainingData):
+def trainModel(model, trainingData, labelRefs, verbosity=0):
   """
   Train the given model on trainingData. Return the trained model instance.
   """
-
-  # Do three passes if we're using a TM.
+  # Do three passes if we're using a TM
   numPasses = 1
   if isinstance(model, ClassificationModelHTM) and (model.tmRegion is not None):
     numPasses = 3
 
+  modelName = repr(model).split()[0].split(".")[-1]
   print
-  print "=======================Training model on sample text================"
-  for passes in range(numPasses):
-    for docId, doc in enumerate(trainingData):
-      document = doc[0]
-      labels = doc[1]
-      print
-      print "Document=", document, "label=",doc[1], "id=",docId
+  print "==============Training {} on sample text============".format(modelName)
+  if verbosity > 0:
+    printTemplate = PrettyTable(["ID", "Document", "Label"])
+    printTemplate.align = "l"
+    printTemplate.header_style = "upper"
+  for _ in xrange(numPasses):
+    for (document, labels, docId) in tqdm(trainingData):
+      if verbosity > 0:
+        docStr = unicode(document, errors="ignore")
+        printTemplate.add_row(
+          [docId, wrapper.fill(docStr), labelRefs[labels[0]]])
       model.trainDocument(document, labels, docId)
+    if verbosity > 0:
+      print printTemplate
 
   return model
 
 
-def testModel(args, model, testData):
+
+def testModel(model, testData, labelRefs, verbosity=0):
   """
-  Test the given model on testData and print out accuracy.
+  Test the given model on testData, print out and return accuracy percentage.
 
   Accuracy is calculated as follows. Each token in a document votes for a single
-  category. The document is classified with the category that received the most
-  votes. Note that it is possible for a token and/or document to receive no
-  votes, in which case it is counted as a misclassification.
+  category; it's possible for a token to contribute no votes. The document is
+  classified with the category that received the most votes. Note that it is
+  possible for a document to receive no votes, in which case it is counted as a
+  misclassification.
   """
+  modelName = repr(model).split()[0].split(".")[-1]
   print
-  print "==========================Classifying sample text================"
-  numCorrect = 0
-  for docId, doc in enumerate(testData):
-    document = doc[0]
-    desiredLabels = doc[1]
+  print "==============Testing {} on sample text=============".format(modelName)
+  if verbosity > 0:
     print
-    print "Document=", document,", desired label: ",desiredLabels
+    printTemplate = PrettyTable(
```

Comment:
All these lines are getting in the way of understanding the core API. We don't need them in this example.